lib/features/email/data/network/email_api.dart

```diff
@@ -158,19 +162,49 @@ class EmailAPI {
       setEmailInvocation.methodCallId,
       SetEmailResponse.deserialize);
 
-    final setEmailSubmissionResponse = response.parse<SetEmailResponse>(
+    final setEmailSubmissionResponse = response.parse<SetEmailSubmissionResponse>(
       setEmailSubmissionInvocation.methodCallId,
-      SetEmailResponse.deserialize,
+      SetEmailSubmissionResponse.deserialize,
       methodName: setEmailInvocation.methodName);
 
     final emailCreated = setEmailResponse?.created?[emailNeedsToBeCreated.id.id];
-    if (emailCreated != null) {
-      return setEmailSubmissionResponse?.notUpdated == null &&
-          setEmailSubmissionResponse?.notCreated == null &&
-          setEmailSubmissionResponse?.notDestroyed == null;
-    } else {
-      return false;
+
+    final mapErrors = _handleSendEmailResponse(
+      response: emailCreated != null ? setEmailResponse : null,
+      submissionResponse: setEmailSubmissionResponse
+    );
+
+    return Map.fromEntries(mapErrors);
+  }
+
+  List<MapEntry<Id, SetError>> _handleSendEmailResponse({
+    SetEmailResponse? response,
+    SetEmailSubmissionResponse? submissionResponse
+  }) {
+    final List<MapEntry<Id, SetError>> remainedErrors = [];
+    if (response != null) {
+      handleSetErrors(
+        notDestroyedError: response.notDestroyed,
+        notUpdatedError: response.notUpdated,
+        notCreatedError: response.notCreated,
+        unCatchErrorHandler: (setErrorEntry) {
+          remainedErrors.add(setErrorEntry);
+          return false;
+        }
+      );
```

Comment:
why not iterate each item in `response.notDestroyed`, `response.notUpdated`, `response.notCreated` then add to `remainedErrors`?